src/Services/NodePastDealsService.ts

```diff
@@ -28,4 +29,22 @@ export class NodePastDealsService {
         }
         return await NodePastDealModel.create({cid, state, size, provider, price, duration, nodeId});
     }
+
+    public async fetchNodePastDeals(userId: number, from: number, to: number, orderBy: string) {
+        return await NodePastDealModel.findAll({
+            attributes: ['id', 'cid', 'state', 'size', 'provider',
+                'price', 'duration', 'createdAt', 'updatedAt'],
+            include: [{
```

Comment:
No need for the include, we don't use it in other endpoints either so this shouldn't be a special case

Reply:
The idea was to retrieve only past deals which belong to a node that belongs to the currently logged in user. Little bit confusing syntax, but the only way to achieve that.